pipeline/lib/json_stream.ml

```diff
@@ -125,9 +174,10 @@ module Save = struct
     let run_job_id = Some worker_job_id in
     let json_stream = job_output_stream worker_job_id in
     Lwt_stream.fold
-      (fun jsons acc ->
-        let jsons = String.concat "\n" jsons in
-        let jsons = Util.parse_jsons jsons in
+      (fun (jsons, line_ranges) acc ->
+        let jsons = jsons |> List.rev |> List.map Yojson.Safe.from_string in
```

Comment:
As our parser doesn't guarantee valid json (only well formed brackets of `{}[]""`), we should probably catch and filter out the values that Yojson rejects :)

Reply:
Should we filter it out or let Yojson crash? It could be invalid non-JSON that our parser thinks is JSON or incorrect JSON generated by our users too, in which case having Yojson crash would be useful to debug builds where benchmarks don't show up.

Reply:
Right... but it's more likely that our parser gets confused and selects "invalid json" that yojson will reject, than users producing "invalid json that's still recognized by our parser but not by yojson". As long as we still signal the schema errors in the json raised by `Current_bench_json`, it should be fine to treat the Yojson failures as "a mistake in our parser" (?)

Reply:
Yes, that makes sense. I guess users would use a valid JSON library to produce their JSON and the errors would be in the schema, if at all, rather than the structure of the JSON.

Reply:
This is not a change from this PR. So, if it seems reasonable to you, let's add this in a separate PR?

Reply:
Nevermind, I pushed a couple of commits on this branch to handle "incorrect" JSON parsed by our parser and rejected by Yojson.